app_integrations/apps/app_base.py

```diff
@@ -184,21 +191,64 @@ def _initialize(self):
         return True
 
     def _finalize(self):
-        """Method for performing any final steps, like saving applicable state"""
-        self._config.mark_success()
+        """Method for performing any final steps, like saving applicable state
 
+        This function is also used to invoke a new copy of this lambda in the case
+        that there are more logs available to collect.
+        """
         if not self._last_timestamp:
             LOGGER.error('Ending last timestamp is 0. This should not happen and is likely '
                          'due to the subclass not setting this value.')
 
         if self._last_timestamp == self._config.start_last_timestamp:
-            LOGGER.error('Ending last timestamp is the same as the beginning last timestamp')
+            LOGGER.error('Ending last timestamp is the same as the beginning last timestamp. '
+                         'This could occur if there were no logs collected for this execution.')
 
         LOGGER.info('App complete for service \'%s\'. Gathered %d logs in %d polls.',
                     self.type(), self._gathered_log_count, self._poll_count)
 
         self._config.last_timestamp = self._last_timestamp
 
+        # If there are more logs to poll, invoke this app function again and do not
+        # mark the config as 'success' (keeping the state as 'running' prevents
+        # scheduled function invocations from running alongside this invocation).
+        if self._more_to_poll:
+            self._invoke_successive_gather()
+            return
+
+        self._config.mark_success()
+
+    def _invoke_successive_gather(self):
+        """Method for invoking a successive app function to handle more logs
```

Comment:
I don't think you need to mention it's a method, that's a little too `meta`

Reply:
Weird nit but I'll change it